contrib/plugs/plug_dependencies.py

```diff
@@ -0,0 +1,137 @@
+"""Extension to OpenHTF to provide dependency declaration for Plugs.
+
+To declare a dependency, add an attribute to the dependent class with the
+PlugDependency descriptor:
+  class NeededPlug(plugs.BasePlug):
+    pass
+
+  class AnotherPlug(plugs.BasePlug):
+    needed = PlugDependency(NeededPlug)
+
+The depended upon plug is then available after the initialization takes place,
+which is done by the phase created by create_dependency_init_phase.
+
+The dependent plugs must not expect their dependencies to work during their
+__init__ calls.
+
+This makes no guarantee about when plugs are torn down; therefore, dependent
+plugs must not use their dependencies during their own tear down.
```

Comment:
Not sure if it's just me, but I find the use of depend* in naming and docstring to be really confusing. i.e. "depended upon", "dependent plugs", "their dependencies".. I get lost in the grammar and have to think too much about the relationship.  
Would it make sense to change naming to use a more concrete relationship like parent/child?

Reply:
I'm leaning towards Farz's suggestion of a dependency attribute, which would then set attributes at initialization time.